### PR TITLE
Add CLIP image conditioning fusion

### PR DESCRIPTION
**Pull request by silveroxides**
## Summary

Adds `CLIPTextEncodeImageFusion`, a generic conditioning node that encodes each image separately and spatially interleaves the resulting visual tokens.

Supported methods:

- Checkerboard
- Block interleave
- Deterministic random dither

Images are passed to the loaded tokenizer at their original resolution. When sources produce different visual grids, their encoded visual features are interpolated in 2D onto the first image's grid before fusion. The node does not create VAE reference latents.

The initial supported encoders are the multimodal CLIP implementations used by Krea 2 and Ideogram 4. Prompt templates remain owned by the loaded tokenizer.

The implementation was adapted from the Visual Component Fusion prototype in [ComfyUI-UtilsCollection](https://github.com/silveroxides/ComfyUI-UtilsCollection).

## Test inputs

<table>
  <tr>
    <th>Input 1</th>
    <th>Input 2</th>
    <th>Input 3</th>
  </tr>
  <tr>
    <td>Input 1</td>
    <td>Input 2</td>
    <td>Input 3</td>
  </tr>
</table>

### Workflow

[SpatialFusion_Krea2WF_New.json](https://github.com/user-attachments/files/30062866/SpatialFusion_Krea2WF_New.json)

### Result

ComfyUI_temp_luepy_00024_

Tested with the Qwen3-VL 4B encoder. Spatial fusion affected the generated result using all three input images.

### Workflow

[SpatialFusion_IdeogramWF_New.json](https://github.com/user-attachments/files/30063780/SpatialFusion_IdeogramWF_New.json)

### Result

ComfyUI_temp_fmkru_00002_

Tested with the Qwen3-VL 8B encoder and Ideogram 4 layered conditioning. Spatial fusion affected the generated result, although Ideogram 4 provides less prompt flexibility. EDIT: Caveat of Ideogram 4 is that it requires the input images to be low enough size to not interfer with the overall conditioning. New updated workflow demonstrates this by resizing the images to equivalent of 384x384 or 0.1475 megapixels.

## Tests

- Covers all three spatial masks and deterministic seeds.
- Covers unequal visual-grid interpolation and first-image grid ownership.
- Covers Krea-style prefix stripping and unstripped token layouts.
- Covers dtype, conditioning metadata, batched image ordering, and unchanged tokenizer inputs.
- 15 focused unit tests pass.
- Ruff passes on changed Python files.

### CODE REVIEW

File: tests-unit/comfy_extras_test/test_qwen_visual_fusion.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import torch
+
+from comfy.cli_args import args as cli_args
+
+if not torch.cuda.is_available():
+    cli_args.cpu = True
```
[the remark this answers is not in the document]
Comment (silveroxides) [reply tests-unit/comfy_extras_test/test_qwen_visual_fusion.py:8]: Suggestion does not solve the CI failure. Pytest fixtures run after the test module is imported. The CUDA failure happens while importing nodes_qwen.py. Newer commit resolves this conversation.

File: comfy/text_encoders/krea2.py [hunk 2]
```diff
@@ -20,6 +20,20 @@
 KREA2_TEMPLATE = "<|im_start|>system\nDescribe the image by detailing the color, shape, size, texture, quantity, text, spatial relationships of the objects and background:<|im_end|>\n<|im_start|>user\n{}<|im_end|>\n<|im_start|>assistant\n"
 
 
+def _krea2_template_end(tok_pairs):
```
Comment (comfyanonymous) [inline comfy/text_encoders/krea2.py]: can you keep the exact same logic as before?

Comment (silveroxides) [reply comfy/text_encoders/krea2.py]: That is gonna be some tricky juggling but I will see if it can be resolved. With normal tokenizer usage, the old logic counts `<|im_start|>` positions using the expanded encoded sequence. Image embeddings shift the later positions, so it calculates the template end after the visual tokens and strips them. This caused “Could not locate the visual token block” failure. So simply reverting krea2.py would break fusion again.

Comment (silveroxides) [reply comfy/text_encoders/krea2.py]: It should be resolved now and I did test using the example workflow included in PR.
